src/opf/opf.jl

```diff
@@ -1,10 +1,244 @@
 using MathOptSymbolicAD
+using SparseArrays
 
 mutable struct OPFModel{OPF <: PM.AbstractPowerModel}
     data::Dict{String,Any}
     model::JuMP.GenericModel
 end
 
+struct OPFData
+    case::String
```

Comment:
Track `base_mva` for per-unit -- nominal conversions.
Not needed for building the OPF models, but useful for downstream applications.

```suggestion
    case::String
    base_mva::Float64
```